realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -190,6 +190,8 @@ public void testGetInstanceClearsCacheWhenFailed() {
 
     public void testInstanceIdForHashCollision() {
         // Ea.hashCode() == FB.hashCode()
+        Realm.deleteRealmFile(getContext(), "Ea");
```

Comment:
It's a bit scary this is needed. I agree it makes a better test case as state is reset, I am just wondering what bug this exposed?